src/core/span_type.hpp

```diff
@@ -84,7 +84,7 @@ class span {
 
 	constexpr size_t size() const noexcept { return static_cast<size_t>( last - first ); }
 	constexpr std::ptrdiff_t ssize() const noexcept { return static_cast<std::ptrdiff_t>( last - first ); }
-	constexpr bool empty() const noexcept { return size() == 0; }
+	constexpr bool empty() const noexcept { return empty(); }
```

Comment:
Don't think so. The lack of `this->` is perhaps hiding this a little.

Reply:
Nice catch, totally overlooked that when looking at these (sed) changes myself.